dolphinscheduler-python/pydolphinscheduler/src/pydolphinscheduler/tasks/python.py

```diff
@@ -55,48 +55,51 @@ def foo():
         want to execute.
     """
 
-    _task_custom_attr = {
-        "raw_script",
-    }
+    _task_custom_attr = {"raw_script", "definition"}
+
+    ext: set = {".py"}
+    ext_attr: Union[str, types.FunctionType] = "_definition"
 
     def __init__(
         self, name: str, definition: Union[str, types.FunctionType], *args, **kwargs
     ):
+        self._definition = definition
         super().__init__(name, TaskType.PYTHON, *args, **kwargs)
-        self.definition = definition
 
     def _build_exe_str(self) -> str:
         """Build executable string from given definition.
 
         Attribute ``self.definition`` almost is a function, we need to call this function after parsing it
         to string. The easier way to call a function is using syntax ``func()`` and we use it to call it too.
         """
-        if isinstance(self.definition, types.FunctionType):
-            py_function = inspect.getsource(self.definition)
-            func_str = f"{py_function}{self.definition.__name__}()"
+        definition = getattr(self, "definition")
+        if isinstance(definition, types.FunctionType):
+            py_function = inspect.getsource(definition)
+            func_str = f"{py_function}{definition.__name__}()"
```

Comment:
we can directly use `self.definition` here

Reply:
It is same as `definition = getattr(self, "definition")`

Reply:
we do you change those, does it raise error?

Reply:
If the `self.definition` is used directly, a warning of `unresolved attribute` will be reported, although it can also run normally.

In init fun, we change the `definition` attribute to `_definition`

Through the `getattr` method, the `definition` attribute will not have a warning.